### PR TITLE
Update driver station for 2018

### PR DESCRIPTION
**Pull request by ArthurAllshire**

### CODE REVIEW

File: wpilib/wpilib/driverstation.py [hunk 1]
```diff
@@ -425,46 +419,44 @@ def isDSAttached(self):
         with self.controlWordMutex:
             self._updateControlWord(False)
             return self.controlWordCache.dsAttached != 0
-    
+
     def isNewControlData(self):
-        """Has a new control packet from the driver station arrived since the
-        last time this function was called?
+        """Gets if a new control packet from the driver station arrived since
```
Comment (Twinters007) [inline wpilib/wpilib/driverstation.py]: Why the change?

File: wpilib/wpilib/driverstation.py [hunk 2]
```diff
@@ -425,46 +419,44 @@ def isDSAttached(self):
...
-            return result
-    
+        return hal.isNewControlData()
+
     def isFMSAttached(self):
-        """Is the driver station attached to a Field Management System?
+        """Gets if the driver station attached to a Field Management System.
 
-        :returns: True if the robot is competing on a field being controlled
+        :returns: true if the robot is competing on a field being controlled
```
Comment (Twinters007) [inline wpilib/wpilib/driverstation.py]: in python `True` is capitalized.

File: wpilib/wpilib/driverstation.py [hunk 3]
```diff
@@ -498,22 +490,16 @@ def getLocation(self):
             return 3
         else:
             return 0
-    
-    def waitForData(self, timeout = None):
+
+    def waitForData(self, timeout = 0):
```
Comment (Twinters007) [inline wpilib/wpilib/driverstation.py]: This actually should be None. 

https://docs.python.org/3.6/library/threading.html#threading.Condition.wait

Comment (ArthurAllshire) [reply wpilib/wpilib/driverstation.py]: Oh, oops, fixed.

File: wpilib/wpilib/driverstation.py (hunk 2 above)
Comment (ArthurAllshire) [reply wpilib/wpilib/driverstation.py]: `True`
Fixed

File: wpilib/wpilib/driverstation.py (hunk 1 above)
Comment (ArthurAllshire) [reply wpilib/wpilib/driverstation.py]: It changed in WPILib.

**Comment by virtuald** [comment]
The HAL changes are in, you should rebase and see if the tests pass.

**Comment by virtuald** [comment]
I fixed pyfrc, tests fail with your bug now.

**Comment by auscompgeek** [comment]
help I'm trying to update DriverStation again but @ArthurAllshire didn't update the git-source-track line and now I'm looking at a long list of changes which have already been made and this is making me cry

**Comment by ariovistus** [comment]
looks like git track was set to a6e6ae41b95caf45c2fd7a1ed4dde5420b31802b at the time
